frontend/src/views/account-mgmt/FacilityView.vue

```diff
@@ -0,0 +1,271 @@
+                <v-col>
+                  <AppLabel>Current primary contact info:</AppLabel>
+                </v-col>
+                <v-col>
+                  <v-row no-gutters justify="end">
+                    <v-icon v-if="isEditablePrimaryContact" icon="fa:fa-regular fa-edit" class="mr-4"></v-icon>
+                  </v-row>
+                </v-col>
+              </v-row>
+              <v-row v-for="(item, index) in this.primaryContact" :key="index">
+                <v-col cols="auto">
+                  <AppLabel>Name: </AppLabel>
+                </v-col>
+                <v-col cols="2">
+                  {{ item.firstName }} {{ item.lastName }}
+                </v-col>
+                <v-col cols="auto">
+                  <AppLabel>Role:</AppLabel>
+                </v-col>
+                <v-col cols="auto">
+                  {{ getRoleNameById(item.role) }}
+                </v-col>
+              </v-row>
+            </div>
+          </v-skeleton-loader>
+        </v-card>
+      </v-col>
+    </v-row>
+    <v-row>
+      <v-col cols="10">
+        <v-row>
+          <AppButton id="back-to-account-mgmt" :primary="false" size="medium" width="400px" :to="{ name: 'account-mgmt' }" :loading="loading">&larr; Back to Account Management</AppButton>
+        </v-row>
+      </v-col>
+      <v-col cols="2">
+        <v-row>
+          <AppButton id="cancel" :primary="true" size="large" :to="{ name: 'organization-facility' }" :loading="loading" class="mr-6">Cancel</AppButton>
+          <AppButton id="save" :disabled="true" :primary="true" size="large" :to="{ name: 'account-mgmt' }" :loading="loading">Save</AppButton>
+        </v-row>
+      </v-col>
+    </v-row>
+  </v-container>
+</template>
+
+<script>
+import AppButton from '@/components/ui/AppButton.vue'
+import AppLabel from '@/components/ui/AppLabel.vue'
+import FacilityInfo from '@/components/facilities/FacilityInfo.vue'
+import FacilityService from '@/services/facilityService'
+import alertMixin from '@/mixins/alertMixin'
+import { useAppStore } from '@/stores/app'
+import { mapState } from 'pinia'
+import { useAuthStore } from '@/stores/auth'
+
+export default {
+  name: 'FacilityView',
+  components: { AppButton, AppLabel, FacilityInfo },
+  mixins: [alertMixin],
+  data() {
+    return {
+      facilityId: null,
+      licences: undefined,
+      contacts: undefined,
+      loading: false,
+    }
+  },
+  computed: {
+    ...mapState(useAuthStore, ['userInfo']),
+    ...mapState(useAppStore, ['healthAuthorities']),
+    expenseAuthorities() {
+      return this.contacts?.filter(contact => contact.isExpenseAuthority)
+    },
+    primaryContact() {
+      return this.contacts?.filter(contact => contact.isPrimaryContact)
+    },
+    isEditablePrimaryContact() {
+      return this.primaryContact && this.primaryContact.length !== 0
+    },
+    isEditableExpenseAuthority() {
+      return this.expenseAuthorities && this.expenseAuthorities.length !== 0
```

Comment:
Would optional chaining (? operator) work here instead of the initial check?

Reply:
updated